File: vpbuf/examples/poker/rel2/nodejs/src/stack.js
#!/usr/bin/nodejs
// Software released under the MIT license (see project root for license file)

"use strict";

const persist = require("./persist");

// ---------------------------------------------------------------------

(function() {
    var library = {
        build_poker_deck: function(max_count) {
            var d = new persist.Deck()
            var count = 0

            for (const i of persist.suit_arr) {
                for (const j of persist.rank_arr) {
                    if (++count > max_count)
                        return d
                    var c = new persist.Card('' + j + ' of ' + i, count)
                    d.cards.push(c)
                }
            }
            return d
        }
    };

    // -----------------------------------------------------------------
    // begin command line processing

    const args = process.argv.slice(2)

    if (args.length == 0) {
        console.log('usage is: stack <deck_name> [ <card_count> ]')
    } else {
        var deck_name = args[0]
        var count = 3

        if (args.length > 1)
            count = args[1]

        console.log('stack ' + count + ' cards')

        var deck = library.build_poker_deck(count)

        persist.save_deck(deck_name, deck)
    }
})();

// ---------------------------------------------------------------------

File: vpbuf/examples/poker/rel2/python/stack.py
#!/usr/bin/python3

# Software released under the MIT license (see project root for license file)

from pyamf import amf3
from vp_poker.persist import *
from vp_poker.vp_poker import *
from vp_poker.tools import *

import argparse

# ------------------------------------------------------------------------------

def build_poker_deck(card_count):
    deck = Deck()

    count = 0
    for s in suit_arr:
        for r in rank_arr:
            count += 1
            if (count > card_count):
                return deck;
            deck.cards.append(Card(r + " of " + s, count))

    return deck

# ------------------------------------------------------------------------------

def main():
    # process args

    p = argparse.ArgumentParser()
    p.add_argument("deck_name")
    p.add_argument("count", nargs='?', default=3)

    args = p.parse_args()

    print('[vrange=',get_low_version(),':',get_high_version(),
            ', python] STACK ', args.count, ' cards', sep='')

    deck = build_poker_deck(int(args.count))

    save_deck(args.deck_name, deck)

if __name__ == '__main__':
    main()

# ------------------------------------------------------------------------------